org.lflang/src/org/lflang/generator/GeneratorCommandFactory.java

```diff
@@ -0,0 +1,144 @@
+ are permitted provided that the following conditions are met:
+
+ 1. Redistributions of source code must retain the above copyright notice,
+ this list of conditions and the following disclaimer.
+
+ 2. Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation
+ and/or other materials provided with the distribution.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ***************/
+
+package org.lflang.generator;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.lflang.ErrorReporter;
+import org.lflang.FileConfig;
+import org.lflang.util.LFCommand;
+
+/**
+ * A factory class responsible for creating commands for use by the LF code generators.
+ * <p>
+ * In addition to the basic functionality of LFCommand, this class additionally ensures that error messages (or
+ * optionally warnings) are shown when a command is not found and that certain environment variables are set (see
+ * {@link #createCommand(String, List, Path, Boolean)} createCommand}).
+ */
+public class GeneratorCommandFactory {
+
+    protected final ErrorReporter errorReporter;
+    protected final FileConfig fileConfig;
+
+
+    /**
+     * Constructor
+     */
+    public GeneratorCommandFactory(ErrorReporter errorReporter, FileConfig fileConfig) {
+        assert errorReporter != null && fileConfig != null;
+        this.errorReporter = errorReporter;
+        this.fileConfig = fileConfig;
+    }
+
+
+    /**
+     * Create a LFCommand instance form a given command and an argument list.
+     * <p>
+     * The command will be executed in the CWD and if the command cannot be found an error message is shown.
+     *
+     * @param cmd  the command to look up
+     * @param args a list of arguments
+     * @return an LFCommand object or null if the command could not be found
+     * @see #createCommand(String, List, Path, Boolean)
+     */
+    public LFCommand createCommand(String cmd, List<String> args) {
+        return createCommand(cmd, args, null, true);
+    }
+
+
+    /**
+     * Create a LFCommand instance form a given command, an argument list and a directory.
```

Comment:
"form" -> "from"
Also, this error is repeated throughout the file.

Reply:
The wonders of copy paste... Thanks for pointing it out!